sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TableLookupCacheSuite.scala

```diff
@@ -46,7 +48,12 @@ class TableLookupCacheSuite extends AnalysisTest with Matchers {
       ignoreIfExists = false)
     val v2Catalog = new InMemoryTableCatalog {
       override def loadTable(ident: Identifier): Table = {
-        V1Table(externalCatalog.getTable("default", ident.name))
+        val catalogTable = externalCatalog.getTable("default", ident.name)
+        new InMemoryTable(
+          catalogTable.identifier.table,
+          catalogTable.schema,
+          Array.empty,
+          Map.empty[String, String].asJava)
```

Comment:
change V1Table to V2Table here, because to lookup V1 table, the catalog will return `UnresolvedCatalogRelation` first and it needs `FindDataSourceTable` rule to resolve. But, `FindDataSourceTable` is in sql/core while the test is in sql/catalyst. So we need to use V2Table instead to avoid depending on `FindDataSourceTable`